src/Campaigns/Routes/RegisterCampaignRoutes.php

```diff
@@ -231,7 +231,7 @@ public function getSchema(): array
                     'description' => esc_html__('Campaign goal type', 'give'),
                 ],
                 'defaultFormId' => [
-                    'type' => 'integer',
+                    'type' => 'number',
```

Comment:
There is any specific reason to use `number` instead of `integer` here?

Reply:
My understanding is that `reactHookForms` is validating as a `number`. When I used `integer` it would throw an error. Likewise, the `goal` property of the schema uses `number`. The `id` property uses `integer`, but the `id` isn't being validated with `reactHookForms`.

Reply:
@kjohnson In this case, I think the best approach is to create a `react-hook-form` custom rule in the fronted to validate integer numbers because changing this parameter here to `number` will not reflect what exactly this endpoint accepts as a parameter. If someone using our API uses the `OPTIONS` to retrieve the parameters schema for this endpoint, they will see that the `id` can be a `number` which is wrong since it can only be an `integer` value. Also, if we decide to provide some documentation for our API using a plugin like the `wp-api-swaggerui` the docs generated by it will display the wrong type for this parameter as well.